### DISCUSSION TITLE
Using a variable when traversing an array with yq causes strange behaviour

### DISCUSSION THREAD

**Discussion by klimenkov** [mikefarah/yq#1565] [Q&A]
Consider an example: `echo [1, 2] | yq '[.[] | . as $x | $x + 1]'`
it returns a list of 4 elements:
```
- 2
- 3
- 2
- 3
```
but I expected a list of 2 values `[2, 3]`.
Is it a feature of _product_ traversal? If so, how to make it _flat_?

**Comment by mikefarah** [comment]
Yeah that's a bug :( will lok at when I get a chance

**Reply by mikefarah** [reply]
Created bug: https://github.com/mikefarah/yq/issues/1566 will track the issue there.

**Reply by mikefarah** [reply]
Fixed in v4.31.2
